### PR TITLE
Add the auto-feature for Smart mode

### PR DESCRIPTION
**Pull request by dajenacuko**
## Abstract
Mesh supports now the auto value in height or width. The changes are done in the Image Manipulation and in BinaryFieldResponse. The height and width are now strings.

REDUNDANT: Opened a new one: https://github.com/gentics/mesh/pull/1055
### General

* [x] Added abstract that describes the change
* [x] Added changelog entry to `/CHANGELOG.adoc`
* [ ] Ensured that the change is covered by tests
* [ ] Ensured that the change is documented in the docs

### On API Changes

* [ ] Checked if the changes are breaking or not
* [ ] Added GraphQL API if applicable
* [ ] Added Elasticsearch mapping if applicable

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -31,6 +31,11 @@ icon:check[] Core: Fixed a bug that caused an internal server error if a project
 
 icon:plus[] Test: An enhanced test container class has been added. The `MeshContainer` class replaces the `MeshDockerServer` which was deprecated.
 
+[[v1.4.5]]
+== 1.4.5 (15.04.2020)
+
+icon:plus[] Image Manipulation: Mesh supports now the ´auto´ parameter when manipulating the images. The width and height fields in the binaries are now strings.
```
Comment (philippguertler) [inline CHANGELOG.adoc:37]:
```suggestion
icon:plus[] Image Manipulation: Gentics Mesh now supports the ´auto´ parameter when manipulating images.
```

Comment (philippguertler) [reply CHANGELOG.adoc:37]: Also provide a link to the documentation

File: core/src/main/java/com/gentics/mesh/core/endpoint/node/BinaryFieldResponseHandler.java [hunk 2]
```diff
@@ -130,22 +129,29 @@ private void resizeAndRespond(RoutingContext rc, BinaryGraphField binaryField, I
 				imageParams.setFocalPoint(fp);
 			}
 		}
+		Integer originalHeight = binaryField.getBinary().getImageHeight();
+		Integer originalWidth = binaryField.getBinary().getImageWidth();
+
+		if (imageParams.getHeight().equals("auto")) imageParams.setHeight(originalHeight);
+		if (imageParams.getWidth().equals("auto")) imageParams.setWidth(originalWidth);
```
Comment (philippguertler) [inline core/src/main/java/com/gentics/mesh/core/endpoint/node/BinaryFieldResponseHandler.java:136]: Please always use braces for `if`
https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces

File: doc/src/main/docs/image-manipulation.asciidoc [hunk 3]
```diff
@@ -34,7 +34,8 @@ image::https://demo.getmesh.io{apiLatest}/demo/webroot/images/ford-gt.jpg?w=500[
 When loading an image with resizing parameters you can provide either the desired width by appending `w=:width` or the desired height by appending by appending `h=:height`. 
 
 ==== Smart resizing and Cropping
-It also possible to provide both dimensions. Note that in this case Mesh potentially crops your image to fit the desired dimensions. This happens when the aspect ratio of the source image is different from the requested dimensions. In these cases Mesh will crop _either_ pixels at the top and bottom or left and right.
+It also possible to provide both dimensions. Note that in this case Mesh potentially crops your image to fit the desired dimensions. This happens when the aspect ratio of the source image is different from the requested dimensions. In these cases Mesh will crop _either_ pixels at the top and bottom or left and right. Mesh supports the `auto` dimension when manipulating an image.
+If `auto` is used in width or height then mesh will crop the use the original value when cropping the image.
```
Comment (philippguertler) [inline doc/src/main/docs/image-manipulation.asciidoc:38]:
```suggestion
If `auto` is used in width or height then mesh will use the original value when cropping the image.
```

File: core/src/test/java/com/gentics/mesh/core/node/NodeImageResizeEndpointTest.java [hunk 4]
```diff
@@ -220,6 +214,94 @@ public void testTransformImageResizeSmart() throws Exception {
...
+		String version = uploadImage(node, "en", "image").getVersion();
+		MeshCoreAssertion.assertThat(testContext).hasUploads(1, 1).hasTempFiles(0).hasTempUploads(0);
+
+		// 2. Transform the image
+		ImageManipulationParameters params = new ImageManipulationParametersImpl();
+		params.setWidth("auto");
+		params.setHeight(200);
+
+		NodeResponse transformResponse = call(() -> client().transformNodeBinaryField(PROJECT_NAME, uuid, "en", version, "image", params));
+		assertEquals("The image should have been resized", 200, transformResponse.getFields().getBinaryField("image").getHeight().intValue());
```
[the remark this answers is not in the document]
Comment (dajenacuko) [reply core/src/test/java/com/gentics/mesh/core/node/NodeImageResizeEndpointTest.java:234]: It is in the 4th section on the validateResizeImage, where both are check. But i can make a new assertion :)

[the remark this answers is not in the document]
Comment (philippguertler) [reply core/src/test/java/com/gentics/mesh/core/node/NodeImageResizeEndpointTest.java:234]: Yes but this only checked the downloaded file, not the direct value reported from Mesh, which could be faulty. That's why the additional assertion doesn't hurt.

**Comment by Jotschi** [comment]
PR needs to be based on hotfix-1.4.x in order to be part of Mesh 1.4.x (to be included in CMP).

**Pull request closed without merging**